File: LFA/Projeto/geometrics-lfa-07-master/pygame/Ponto.py
import math
class Ponto:

    def __init__(self, x, y):
        self.x = x
        self.y = y
    
    #setter method to change the value of the x coordinate
    def set_x(self,x):
        self.x = x

    #setter method to change the value of the y coordinate
    def set_y(self,y):
        self.y = y 
    
    #getter method to get x coordinate
    def get_x(self):
        return self.x
    
    #getter method to get y coordinate
    def get_y(self):
        return self.y
    
    #method to add 2 points
    def __add__(self,other):
        self.x += other.x
        self.y += other.y


    ## Estas duas podem dar jeito para a linha principalmente a distance
    #method to find the halfway point
    def __halfWayTo__(self,other):
        return Ponto((self.x + other.x)/2.0,(self.y + other.y)/2.0)


    #method to find the distance between 2 points
    def __distanceTo__(self,other):
        return round(math.sqrt( ((self.x-other.x)**2) + ((self.y-other.y)**2) ))

    # nao é necessario
    def __equals__(self,other):
        if(self.x == other.x and self.y == other.y):
            return True
        else:
            return False

    def __pos__ (self):
        return (self.x, self.y)

    


    

File: 1ºSemestre/Programacao-III/aula3/Ponto.java
package aula3;

public class Ponto {
    private double x, y;

    Ponto() {
        this.x = 0; this.y = 0;
    }

    Ponto(double x, double y) {
        this.x = x; this.y = y;
    }

    public double getX() {
        return x;
    }

    public double getY() {
        return y;
    }

    public void add(Ponto point) {
        this.x += point.x;
        this.y += point.y;
    }

    public void subtract(Ponto point) {
        this.x -= point.x;
        this.y -= point.y;
    }

    public Ponto halfWayTo(Ponto point) {
        return new Ponto((this.x + point.x)/2.0,(this.y + point.y)/2.0);
    }

    public double distanceTo(Ponto point) {
        return Math.sqrt(Math.pow(this.x-point.x,2) + Math.pow(this.y-point.y,2));
    }

    @Override
    public boolean equals(Object obj) {
        if (obj instanceof Ponto) {
            Ponto temp = (Ponto)obj;
            return this.x == temp.x && this.y == temp.y;
        } else {
            return super.equals(obj);
        }
    }

    @Override
    public String toString() {
        return String.format("(%.2f, %.2f)", x, y);
    }
}
